src/v/pandaproxy/client/record_set.cc

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "pandaproxy/client/record_set.h"
+
+#include "kafka/requests/kafka_batch_adapter.h"
+
+namespace pandaproxy::client {
+
+struct mini_header {
+    model::offset base_offset;
+    int32_t record_count;
+    int32_t record_bytes;
+};
+
+mini_header read_mini_header(iobuf_const_parser& in) {
+    const size_t initial_bytes_consumed = in.bytes_consumed();
+
+    auto base_offset = model::offset(in.consume_be_type<int64_t>());
+    auto batch_length = in.consume_be_type<int32_t>();
+    in.skip(
+      sizeof(int32_t)     // partition_leader_epoch
+      + sizeof(int8_t)    // magic
+      + sizeof(int32_t)   // crc
+      + sizeof(int16_t)); // attrs
+    auto last_offset_delta = in.consume_be_type<int32_t>();
+    in.skip(
+      sizeof(int64_t)   // first_timestamp
```

Comment:
couldbe extracted to constexpr constant.

Reply:
Fixed in #515 - pretty sure constant folding can't fail, though.